src/scripts/install.sh

```diff
@@ -1,45 +1,82 @@
-downloadAndRun() {
-    url=$1
-    shift
+#!/usr/bin/env bash
+
+# Exit script if you try to use an uninitialized variable.
+set -o nounset
+
+# Exit script if a statement returns a non-true return value.
+set -o errexit
+
+# Use the error status of the first failure, rather than that of the last item in a pipeline.
+set -o pipefail
+
+sudoIfAvailable() {
     if [[ -x $(command -v sudo) ]]; then
-    curl -sfL $url | sudo -E bash -s -- "$@"
+    sudo -E bash "$@"
     else
-    curl -sfL $url | bash -s -- "$@"
+    bash "$@"
     fi
 }
 
+downloadAndRun() {
+    url=$1
+    shift
+    curl -sfL $url | sudoIfAvailable -s -- "$@"
+}
+
 os=$(uname)
+tmpdir=$(dirname "$(mktemp -u)")
+mpmbaseurl="https://www.mathworks.com/mpm"
 
-# installer does not support the Docker executor type on Linux
-if [[ $os = Linux ]] && awk -F/ '$2 == "docker"' /proc/self/cgroup | read -r; then
-    echo 'The Docker executor type is not supported.'
-    exit 1
+# install system dependencies
+if [[ $os = Linux ]]; then
+    # install MATLAB dependencies
+    release=$(echo "${PARAM_RELEASE}" | grep -ioE "(r[0-9]{4}[a-b]|latest)")
+    downloadAndRun https://ssd.mathworks.com/supportfiles/ci/matlab-deps/v0/install.sh "$release"
+    # install mpm depencencies
+    sudoIfAvailable -c "apt-get install --no-install-recommends --yes \
+        wget \
+        unzip \
+        ca-certificates"
 fi
 
-# install core system dependencies
-if [[ $os = Linux ]]; then
-    downloadAndRun https://ssd.mathworks.com/supportfiles/ci/matlab-deps/v0/install.sh "${PARAM_RELEASE}"
+# set os specific options
+if [[ $os = CYGWIN* || $os = MINGW* || $os = MSYS* ]]; then
+    batchinstalldir='/c/Program Files/matlab-batch'
+    rootdir="$tmpdir/matlab_root"
+    mpmurl="$mpmbaseurl/win64/mpm";
+    mpmsetup="unzip -q $tmpdir/mpm -d $tmpdir"
+    mpmpath="$tmpdir/bin/win64/mpm"
+    rootdir=$(cygpath "$rootdir")
+    mpmpath=$(cygpath "$mpmpath")
+else
+    rootdir="$tmpdir/matlab_root"
+    batchinstalldir='/opt/matlab-batch'
+    mpmurl="$mpmbaseurl/glnxa64/mpm";
+    mpmsetup=""
+    mpmpath="$tmpdir/mpm"
 fi
```

Comment:
Consistent order of variables in each block would help readability here. 

Some of the variables may also be pulled out of these branches if you have a variable containing the mwarch (e.g. win64, glnxa64, maci64). Would help prepare for macOS as well.